src/lib/tasker.ts

```diff
@@ -0,0 +1,262 @@
+  const testResults = await testResultRepo.find({
+    where: { attemptId: attempt.id },
+  });
+  const totalTests = await testRepo.count({
+    where: { isActive: true, uncompressedStatus: TestResultStatus.VALID },
+  });
+  if (testResults.length < totalTests) {
+    return;
+  }
+  let testsPassed = 0;
+  let totalCompressionRatio = 0;
+  for (const result of testResults) {
+    if (result.status === TestResultStatus.VALID) {
+      testsPassed++;
+      totalCompressionRatio += result.compressionRatio || 0;
+    }
+  }
+  const averageCompressionRatio =
+    testsPassed > 0 ? totalCompressionRatio / testsPassed : 0;
+  attempt.status =
+    totalTests > 0 && testsPassed === totalTests
+      ? AttemptStatus.COMPLETED
+      : AttemptStatus.FAILED;
+  attempt.averageCompressionRatio = averageCompressionRatio;
+  await attemptRepo.save(attempt);
+  console.log(`Attempt ${attempt.id} has been fully processed and aggregated.`);
+}
+
+/**
+ * Runs the main tasker process loop.
+ *
+ * This function performs the following steps in a continuous loop:
+ * 1. Evaluates any uncompressed tests that have not yet been processed.
+ * 2. Finds the oldest pending attempt that still has tests to run.
+ * 3. Processes all remaining tests for the identified attempt.
+ * 4. Aggregates the results once the attempt is fully processed.
+ * 5. Waits for a defined interval before starting the next cycle.
+ */
+async function run() {
+  await AppDataSource.initialize();
+  console.log('Tasker started, connected to DB.');
+  while (true) {
+    try {
+      const attemptRepo = AppDataSource.getRepository(Attempt);
+      const testRepo = AppDataSource.getRepository(Test);
+      const testsToEvaluate = await testRepo.find({
+        where: { uncompressedStatus: IsNull(), isActive: true },
+      });
+      if (testsToEvaluate.length > 0) {
+        console.log(
+          `Found ${testsToEvaluate.length} uncompressed tests to evaluate.`
+        );
+        await Promise.all(testsToEvaluate.map(evaluateUncompressedTest));
+      }
+      const attempt = await attemptRepo
+        .createQueryBuilder('attempt')
+        .where(qb => {
+          const subQuery = qb
+            .subQuery()
+            .select('test.id')
+            .from(Test, 'test')
+            .leftJoin('test.testResults', 'tr', 'tr.attempt_id = attempt.id')
+            .where('test.isActive = true')
+            .andWhere('tr.attempt_id IS NULL')
+            .getQuery();
+          return `EXISTS (${subQuery})`;
+        })
+        .andWhere('attempt.status = :status', { status: AttemptStatus.PENDING })
+        .orderBy('attempt.timestamp', 'ASC')
+        .getOne();
+      if (attempt) {
+        await processAttempt(attempt);
+        await aggregateAttemptResults(attempt);
+      }
+    } catch (error) {
+      console.error('An error occurred in the tasker loop:', error);
+    }
+    await new Promise(resolve => setTimeout(resolve, POLL_INTERVAL));
```

Comment:
if there are no tasks to do, sleep POLL_INTERVAL. If there are some, do not sleep.